x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -534,18 +534,29 @@ export class Embeddable
     if (!activeDatasourceId || !adapters?.requests) return;
     const activeDatasource = this.deps.datasourceMap[activeDatasourceId];
     const docDatasourceState = this.savedVis?.state.datasourceStates[activeDatasourceId];
-    const warnings: React.ReactNode[] = [];
-    this.deps.data.search.showWarnings(adapters.requests, (warning) => {
-      const warningMessage = activeDatasource.getSearchWarningMessages?.(
+    const warningsMap: Map<string, Array<string | React.ReactNode>> = new Map();
+    this.deps.data.search.showWarnings(adapters.requests, (warning, meta) => {
+      const { request, response, requestId } = meta;
+
+      const warningMessages = activeDatasource.getSearchWarningMessages?.(
         docDatasourceState,
-        warning
+        warning,
+        request,
+        response
       );
 
-      warnings.push(...(warningMessage || []));
-      if (warningMessage && warningMessage.length) return true;
+      if (warningMessages?.length) {
+        const key = (requestId ?? '') + warning.type + warning.reason?.type ?? '';
+        if (!warningsMap.has(key)) {
+          warningsMap.set(key, warningMessages);
+        }
+        return true;
+      }
+      return false;
     });
-    if (warnings && this.warningDomNode) {
-      render(<Warnings warnings={warnings} />, this.warningDomNode);
+
+    if (warningsMap.size && this.warningDomNode) {
+      render(<Warnings warnings={[...warningsMap.values()].flat()} />, this.warningDomNode);
```

Comment:
Should this be wrapped in a theme provider? The tooltip doesn't currently pick up the dark mode theme.
<img width="1235" alt="Screen Shot 2022-10-25 at 7 57 21 PM" src="https://user-images.githubusercontent.com/315764/197910658-a41d10e8-5a6c-4ac0-ba16-66f15d40155c.png">